src/lightning/fabric/cli.py

```diff
@@ -187,6 +187,19 @@ def _set_env_variables(args: Namespace) -> None:
 
 def _get_num_processes(accelerator: str, devices: str) -> int:
     """Parse the `devices` argument to determine how many processes need to be launched on the current machine."""
+    if accelerator == "auto" or accelerator is None:
```

Comment:
lets not duplicate the logic and use already written functions like:
https://github.com/Lightning-AI/pytorch-lightning/blob/c85660c0da7777c667309a3ef937276353aacc27/src/lightning/pytorch/trainer/connectors/accelerator_connector.py#L333